src/crypto/x509/verify_test.go

```diff
@@ -1836,7 +1836,7 @@ func TestValidHostname(t *testing.T) {
 	}
 }
 
-func generateCert(cn string, isCA bool, issuer *Certificate, issuerKey crypto.PrivateKey) (*Certificate, crypto.PrivateKey, error) {
+func generateCert(cn string, isCA bool, isDC bool, issuer *Certificate, issuerKey crypto.PrivateKey) (*Certificate, crypto.PrivateKey, error) {
```

Comment:
We also should test this extension in intermediates. It shouldn't appear, but what to do about it? I think just ignore it.

Reply:
I added a restriction that if it is `IsCA` the DC extension should not be added not parsed. Let me know what you think..